airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -91,15 +92,36 @@ def connection_checker(self) -> ConnectionChecker:
 
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         self._emit_manifest_debug_message(extra_args={"source_name": self.name, "parsed_config": json.dumps(self._source_config)})
+        stream_configs = self._stream_configs(self._source_config)
 
         source_streams = [
             self._constructor.create_component(
                 DeclarativeStreamModel, stream_config, config, emit_connector_builder_messages=self._emit_connector_builder_messages
             )
-            for stream_config in self._stream_configs(self._source_config)
+            for stream_config in self._initialize_cache_for_parent_streams(deepcopy(stream_configs))
         ]
+
         return source_streams
 
+    @staticmethod
+    def _initialize_cache_for_parent_streams(stream_configs: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
```

Comment:
can you add unit tests for this method. Specifically, I'd expect a test that verifies parent streams are configured to use the cache if and only if another stream depends on them

Reply:
Added